### ISSUE TITLE
Twitter Integration needs to be added to the app.

### ISSUE THREAD

**Issue by sunderhaus** [sunderhaus/Citizen#5]
We had the plan of adding two features that didn't get worked out before the end of the hackathon. Twitter Integration was one of them, and closely related to that was Local Representative based filters on the feed data returned.

What has been the issue with the twitter integration thus far? @kennethkloeppel, was it related to the OAUTH steps to implement it or was it more difficult to get the tweets without authenticating with Twitter?

The filters could be cookies which would mean we wouldn't have to implement anything server side yet to track the user's choices in a profile. However, we seem to really be butting up against that need quickly. Especially if we are to expand on this to perform more features with Twitter.

**Comment by kennethkloeppel** [comment]
Yeah, there is an OAUTH handshake that needs to happen before you can call the search functions.  Once the OAUTH is done, the rest looks straight forward.  We can search by handle as well as hashtag.

I agree we are at a point where we have to decide if we're going to need a way to create a user profile to store the user preferences as to which reps to follow.  Of course this removes a bit of the anonymity.  We could store everything in cookies as you've said, but if something happens to the cookie, the user will have to setup his/her preferences all over again.

**Comment by westown** [comment]
Here are my thoughts on the Twitter integration. Keep in mind that I've received no feedback on this from government folks but plan to soon. I do, however, think that Twitter integration, though complex, will be a defining feature of Citizen: allowing reps to have direct, one-way communication to their constituents using a platform that already exists.

First of all, the alerts to a user without an account would only be possible if they had a downloadable app (right?). With an app, the default settings would be set to notify the user when select government titles in their area – i.e. their council member and state representative – tweet using a specific hashtag. If the user wants alerts on anyone else, then we would request that they create an account – unless there was a way....

If the user is using Citizen on a desktop computer, there would be no preset alerts, but a bell icon would be visible by all of the officials. If the user clicks one of the bells to request receiving an alert, then we would request that they create an account. We would then send the alerts of that title in that district to their email address.

Real time alerts would probably require an engine to keep tabs on the specified hashtag. When a tweet appears on the hashtag stream, the twitter handle would be compared with a list of all of the twitter handles of the state of representatives, council members, and any other official who have an active alert setting. 

If there are any twitter handle matches, then the alert is sent to any device that last reported being in their district, or any user that requested alerts from that official. 

Let me know what you guys think of this and if it all makes sense. This feature probably isn't created until I have confirmation that this would really help officials.
